components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/SystemDefaultClaimMetadataManager.java

```diff
@@ -0,0 +1,274 @@
+
+    private static final List<ClaimDialect> claimDialects;
+    private static final Map<String, List<Claim>> claims;
+
+    static {
+
+        claimDialects = new ArrayList<>();
+        claims = new HashMap<>();
+
+        ClaimConfig claimConfig = IdentityClaimManagementServiceDataHolder.getInstance().getClaimConfig();
+        if (claimConfig != null && claimConfig.getClaimMap() != null) {
+            String primaryDomainName = IdentityUtil.getPrimaryDomainName();
+
+            for (Map.Entry<ClaimKey, ClaimMapping> entry : claimConfig.getClaimMap().entrySet()) {
+                ClaimKey claimKey = entry.getKey();
+                ClaimMapping claimMapping = entry.getValue();
+                String claimDialectURI = claimKey.getDialectUri();
+                String claimURI = claimKey.getClaimUri();
+                Claim claim;
+
+                boolean dialectExists = claimDialects.stream()
+                        .anyMatch(claimDialect -> claimDialect.getClaimDialectURI().equals(claimDialectURI));
+                if (!dialectExists) {
+                    claimDialects.add(new ClaimDialect(claimDialectURI));
+                }
+
+                if (claimDialectURI.equals(LOCAL_CLAIM_DIALECT_URI)) {
+                    List<AttributeMapping> mappedAttributes = new ArrayList<>();
+                    if (StringUtils.isNotBlank(claimMapping.getMappedAttribute())) {
+                        mappedAttributes
+                                .add(new AttributeMapping(primaryDomainName, claimMapping.getMappedAttribute()));
+                    }
+
+                    if (claimMapping.getMappedAttributes() != null) {
+                        for (Map.Entry<String, String> claimMappingEntry : claimMapping.getMappedAttributes()
+                                .entrySet()) {
+                            mappedAttributes.add(new AttributeMapping(claimMappingEntry.getKey(),
+                                    claimMappingEntry.getValue()));
+                        }
+                    }
+
+                    Map<String, String> claimProperties = filterClaimProperties(claimConfig.getPropertyHolderMap()
+                            .get(claimKey));
+                    claim = new LocalClaim(claimURI, mappedAttributes, claimProperties);
+                } else {
+                    String mappedLocalClaimURI = claimConfig.getPropertyHolderMap().get(claimKey).get(ClaimConstants
+                            .MAPPED_LOCAL_CLAIM_PROPERTY);
+                    Map<String, String> claimProperties = filterClaimProperties(claimConfig.getPropertyHolderMap()
+                            .get(claimKey));
+                    claim = new ExternalClaim(claimDialectURI, claimURI, mappedLocalClaimURI, claimProperties);
+                }
+                claims.computeIfAbsent(claimDialectURI, k -> new ArrayList<>());
+                claims.get(claimDialectURI).add(claim);
+            }
+        }
+    }
```

Comment:
let's refactor this. can simplify the nesting / extract out the finding mapped attributes part to a helper method to improve readability.

Reply:
Fixed with b159fa412a5dcb2d19480a24da4d91c582b5a501